src/lib.rs

```diff
@@ -307,7 +307,21 @@ impl Config {
                 // studio build system but instead use makefiles that MinGW can
                 // use to build.
                 if self.generator.is_none() {
-                    cmd.arg("-G").arg("MSYS Makefiles");
+                    // If make.exe isn't found, that means we may be using a MinGW
+                    // toolchain instead of a MSYS2 toolchain. If neither is found,
+                    // the build cannot continue.
+                    let has_msys2 = Command::new("make").spawn().err()
```

Comment:
Could this try to use something like `make --version` instead of `make` which could accidentally try to do some work? Additionally, could this use `.output()` to squelch all output and otherwise just test if the binary exists? (searching `PATH` would also be fine)

Reply:
Sure. I've made both of those changes in the new commit.